FeedStoreChallenge/CoreDataFeedStore.swift

```
@@ -0,0 +1,137 @@
+//
+//  CoreDataFeedStore.swift
+//  FeedStoreChallenge
+//
+//  Created by Eric Garlock on 2/16/21.
+//  Copyright © 2021 Essential Developer. All rights reserved.
+//
+
+import Foundation
+import CoreData
+
+public class CoreDataFeedStore: FeedStore {
+	
+	let container: NSPersistentContainer
+	let context: NSManagedObjectContext
+	
+	public init(storeURL: URL) {
+		
+		let bundle = Bundle(for: CoreDataFeedStore.self)
+		let modelURL = bundle.url(forResource: "CoreDataFeedStore", withExtension: "momd")!
+		let model = NSManagedObjectModel(contentsOf: modelURL)!
+		
+		let description = NSPersistentStoreDescription(url: storeURL)
+		
+		let container = NSPersistentContainer(name: "CoreDataFeedStore", managedObjectModel: model)
+		container.persistentStoreDescriptions = [description]
+		
+		container.loadPersistentStores { _, error in
+			if let error = error {
+				fatalError("\(error)")
```

Errors should be handled without `!` or `fatalError`. If something is wrong, you can throw an error so clients can react accordingly as shown in the lectures. Otherwise, you may have runtime errors (crashes!).